java/core/src/main/java/co/worklytics/psoxy/gateway/impl/oauth/OAuthRefreshTokenSourceAuthStrategy.java

```diff
@@ -217,6 +243,32 @@ public Set<ConfigService.ConfigProperty> getRequiredConfigProperties() {
             }
             return propertyStream.collect(Collectors.toSet());
         }
+
+
+        private Optional<AccessToken> getSharedAccessToken() {
+            Preconditions.checkState(payloadBuilder.useSharedToken(), "Shared token not supported");
+            Optional<String> jsonToken = config.getConfigPropertyAsOptional(ConfigProperty.WRITABLE_ACCESS_TOKEN);
+            if (jsonToken.isEmpty()) {
+                return Optional.empty();
+            } else {
+                try {
+                    return Optional.ofNullable(objectMapper.readerFor(AccessToken.class).readValue(jsonToken.get().getBytes(StandardCharsets.UTF_8)));
+                } catch (IOException e) {
+                    log.log(Level.SEVERE, "Could not parse contents of token into an object", e);
+                    return Optional.empty();
+                }
+            }
+        }
+
+        private void storeSharedAccessToken(@NonNull AccessToken accessToken) {
+            Preconditions.checkState(payloadBuilder.useSharedToken(), "Shared token not supported");
+            try {
+                config.putConfigProperty(ConfigProperty.WRITABLE_ACCESS_TOKEN, objectMapper.writerFor(AccessToken.class).writeValueAsString(accessToken));
+                log.log(Level.INFO, "New token stored in config");
+            } catch (JsonProcessingException e) {
+                log.log(Level.SEVERE, "Could not write token into JSON", e);
```

Comment:
for clarity, `Could not serialize token as JSON` (just because this is distinct case from other IO involved in overall "write" operation